include/clad/Differentiator/KokkosBuiltins.h

```diff
@@ -0,0 +1,74 @@
+// This header file contains the implementation of the Kokkos framework
+// differentiation support in Clad in the form of custom pushforwards and
+// pullbacks. Please include it manually to use Clad on Kokkos code.
+
+#ifndef CLAD_KOKKOS_BUILTINS_H
+#define CLAD_KOKKOS_BUILTINS_H
+
+#include <Kokkos_Core.hpp>
+#include "clad/Differentiator/Differentiator.h"
+
+namespace clad {
+namespace custom_derivatives {
+namespace class_functions {
+/// Kokkos arrays
+template <typename ArrayT, typename Space>
+clad::ValueAndPushforward<Kokkos::View<ArrayT, Space>,
+                          Kokkos::View<ArrayT, Space>>
+constructor_pushforward(
```

Comment:
```suggestion
template <class DataType, class... ViewParams>
clad::ValueAndPushforward<Kokkos::View<DataType, ViewParams...>,
                          Kokkos::View<Kokkos::View<DataType, ViewParams...>>>
constructor_pushforward(
```

The full set of possible ``Kokkos::View`` template parameters is more than ``<datatype, space>`` or ``<datatype, layout, space>``. There could also be a memory traits type, and everything besides the datatype is optional. What about doing something like this to match all the possible cases? This would also let you merge this specialization with the one on line 32.

Reply:
Good catch @brian-kelley !

Reply:
thanks for the suggestion, updated this.